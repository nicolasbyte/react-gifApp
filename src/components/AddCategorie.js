import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategorie = ( { setCategories } ) => {

    const [categorie, setCategorie] = useState('');

    const addHandle = ( e ) => {
        
        setCategorie( e.target.value )
    }

    const addSumbit = ( e ) => {
        e.preventDefault();

        if ( categorie.trim().length > 2) {
            
            setCategories( cat => [ categorie , ...cat ] );
            setCategorie( '' );
        }


    }
    return (
        
            <form onSubmit = { addSumbit } >
                <p> { categorie } </p>
                <input
                        type     = "text"
                        value    = { categorie }
                        onChange = { addHandle }
                    />
            </form>

    );

};


AddCategorie.propTypes = {

    setCategories : PropTypes.func.isRequired 
}
