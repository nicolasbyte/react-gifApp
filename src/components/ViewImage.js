import React from 'react';
import PropTypes from 'prop-types';


export const ViewImage = ( { title , url } ) => {
    return (
        <div className = "card animate__animated animate__fadeIn">
                <img src = { url } alt = { title } />
                <p> { title } </p>           
        </div>
    );
};


ViewImage.propTypes = {
    title : PropTypes.string.isRequired,
    url   : PropTypes.string.isRequired
};
