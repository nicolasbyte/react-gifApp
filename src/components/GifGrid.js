import React from 'react';
import PropTypes from 'prop-types';

import { useGifFetch } from '../hooks/useGifFetch';
import { ViewImage } from './ViewImage';

export const GifGrid =  ( { category } ) => {

    
    const { loading , data : images } = useGifFetch( category );



    
    

    return (
        <>
        <h3> { category } </h3>
        { loading && <p>Cargando...</p> }
        <div className = "card-grid animate__animated animate__fadeIn">
            
                {
                    images.map(  img  => (
                        <ViewImage 
                                    key = { img.id }
                                    { ...img } />
                    ) )
                }
            
        </div>
        </>
    );
}


GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}