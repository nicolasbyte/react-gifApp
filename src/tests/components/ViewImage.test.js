import React from 'react';
import { shallow } from "enzyme";
import { ViewImage } from '../../components/ViewImage';


describe('pruebas en el componente ViewImage', () => {
    
    const title = 'Un título';
    const url   = 'https://localhost/algo.com';
    
    const wrapper = shallow(<ViewImage title = { title } url = { url }  />);
    test('Fotografia del componente ViewImage', () => {


        expect( wrapper ).toMatchSnapshot();
    });


    test('El titulo debe ser un título', () => {

        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title );
    });
    
    test('La imagen dee tener un src y un alt', () => {
        
        const img = wrapper.find('img');

        expect( img.prop('src')).toBe( url );
        expect( img.prop('alt')).toBe( title );

    });


    test('El div debe tener la clase animate__fadeIn', () => {
        
        const div = wrapper.find('div');

        console.log( div.prop('className'));
        
        expect( div.prop('className').includes('animate__fadeIn')).toBe( true );
    });
    
    
    
});

