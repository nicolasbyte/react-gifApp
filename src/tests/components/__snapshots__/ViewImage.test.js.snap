// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente ViewImage Fotografia del componente ViewImage 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.com"
  />
  <p>
     
    Un título
     
  </p>
</div>
`;
